### ISSUE TITLE
GraphQL filtering clauses using cppgraphqlgen.

### ISSUE THREAD

**Issue by h-barthwal** [microsoft/cppgraphqlgen#233]
I am trying to generate code for a GraphQL schema with the most recent `schemagen` compiler.
The schema is as follows:
```graphql

input IntFilter {                         
  eq: Int                                 
  in: [Int]                               
  le: Int                                 
  lt: Int                                 
  ge: Int                                 
  gt: Int                                 
}                                         
                                          
input StringFilter {                      
  in: [String]                            
  eq: String                              
}                                         
                                          
type Person {                             
  name: String                            
  age: Int                                
  city: String                            
}                                         
                                          
input PersonFilter {                      
  name: StringFilter                      
  age: IntFilter                          
  city: StringFilter                      
  and: PersonFilter                       
  not: PersonFilter                       
  or: PersonFilter                        
}                                         
                                          
type PersonQuery {                        
  persons(filter: PersonFilter): [Person!]
}                                         
                                          
schema {                                  
  query: PersonQuery                      
}                                        
```

I get the following error:
```bash
$ /vcpkg/installed/x64-linux/tools/cppgraphqlgen/schemagen --schema cyclic.graphql --prefix Person --namespace person
Input object cycle type: PersonFilter
```

With cppgraphqlgen version 3.5, I was able to generate a schema that was although not compiled as it is but I was able to modify it to support this use case. For example, it gave me:
```cpp

struct PersonFilter                                                                                                                         
{                                                                                                                                           
        std::optional<StringFilter> name;                                                                                                   
        std::optional<IntFilter> age;                                                                                                       
        std::optional<StringFilter> city;                                                                                                   
        std::optional<PersonFilter> and;                                                                                                    
        std::optional<PersonFilter> not;                                                                                                    
        std::optional<PersonFilter> or;                                                                                                     
};                                                                                                                                          
```
which I was able to modify to
```cpp
struct PersonFilter                                                                                                                         
{                                                                                                                                           
        std::optional<StringFilter> name;                                                                                                   
        std::optional<IntFilter> age;                                                                                                       
        std::optional<StringFilter> city;                                                                                                   
        std::unique_ptr<PersonFilter> and_;                                                                                                    
        std::unique_ptr<PersonFilter> not_;                                                                                                    
        std::unique_ptr<PersonFilter> or_;                                                                                                     
};                                                                                                                                         
```
and then make my app compile and work as expected.  I want to migrate to the newer version of `cppgraphqlgen` but I am not sure how to get around this new behavior introduced in `schemagen`. My questions are:
1. Given that the above schema is valid GraphQL. Does it make sense to support this usecase? 
2. If for some reason this usecase is not supported then can someone suggest a workaround that does not involve forking this library and modifying `schemagen`?

Thanks in advance.

**Comment by wravery** [comment]
The issue in the latest version is #225. The issue that you're seeing in `v3.x` is a matter of not recognizing the alternate C++ keywords/tokens, which was fixed in `main` for #209, but never backported to `v3.x`.

**Comment by wravery** [comment]
Both issues should be fixed now in `main`, please give this another try. Thanks!
